miio/pwzn_relay.py

```diff
@@ -0,0 +1,380 @@
+
+MODEL_PWZN_RELAY_APPLE = 'pwzn.relay.apple'
+
+AVAILABLE_PROPERTIES = {
+    MODEL_PWZN_RELAY_APPLE: [
+        'relay_status', 'on_count', 'name0', 'name1', 'name2', 'name3',
+        'name4', 'name5', 'name6', 'name7', 'name8', 'name9', 'name10',
+        'name11', 'name12', 'name13', 'name14', 'name15'
+    ],
+}
+
+
+class PwznRelayStatus:
+    """Container for status reports from the plug."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response of a PWZN Relay Apple (pwzn.relay.apple)
+        { 'relay_status': 9, 'on_count': 2, 'name0': 'channel1', 'name1': '',
+        'name2': '', 'name3': '', 'name4': '', 'name5': '', 'name6': '',
+        'name7': '', 'name8': '', 'name9': '', 'name10': '', 'name11': '',
+        'name12': '', 'name13': '', 'name14': '', 'name15': '' }
+        """
+        self.data = data
+
+    def relay(self, num: int) -> bool:
+        """Current relay X state."""
+        if "relay_status" in self.data:
+            return True if self.data["relay_status"] & (1 << num) else False
+
+    @property
+    def relay0(self) -> bool:
+        """Current relay0 state."""
+        return self.relay(0)
+
+    @property
+    def relay1(self) -> bool:
+        """Current relay1 state."""
+        return self.relay(1)
+
+    @property
+    def relay2(self) -> bool:
+        """Current relay2 state."""
+        return self.relay(2)
+
+    @property
+    def relay3(self) -> bool:
+        """Current relay3 state."""
+        return self.relay(3)
+
+    @property
+    def relay4(self) -> bool:
+        """Current relay4 state."""
+        return self.relay(4)
+
+    @property
+    def relay5(self) -> bool:
+        """Current relay5 state."""
+        return self.relay(5)
+
+    @property
+    def relay6(self) -> bool:
+        """Current relay6 state."""
+        return self.relay(6)
+
+    @property
+    def relay7(self) -> bool:
+        """Current relay7 state."""
+        return self.relay(7)
+
+    @property
+    def relay8(self) -> bool:
+        """Current relay8 state."""
+        return self.relay(8)
+
+    @property
+    def relay9(self) -> bool:
+        """Current relay9 state."""
+        return self.relay(9)
+
+    @property
+    def relay10(self) -> bool:
+        """Current relay10 state."""
+        return self.relay(10)
+
+    @property
+    def relay11(self) -> bool:
+        """Current relay11 state."""
+        return self.relay(11)
+
+    @property
+    def relay12(self) -> bool:
+        """Current relay12 state."""
+        return self.relay(12)
+
+    @property
+    def relay13(self) -> bool:
+        """Current relay13 state."""
+        return self.relay(13)
+
+    @property
+    def relay14(self) -> bool:
+        """Current relay14 state."""
+        return self.relay(14)
+
+    @property
+    def relay15(self) -> bool:
+        """Current relay15 state."""
+        return self.relay(15)
+
+    @property
+    def name0(self) -> str:
```

Comment:
Instead of having separate methods to return single names, I'd prefer to have `relay_names` or simply `names` returning a dict of (index, name), something like this:
```
def relay_names() -> Dict[int, str]:
    def _extract_index_from_key(name) -> int:
        """extract the index from the variable"""

    return {_extract_index(name): value for name, value in self.data.items() if name.startswith("name")}
```

Reply:
I would be happy about this solution, too.

Reply:
I'm not so familar with python language.
Can you show me more detail about how to fix this?

Reply:
Remove all nameX methods/properties and implement a single method as suggested by @rytilahti. This method should return a Dict with relay names. Do you need additional support to implement this?